plugins/node/opentelemetry-instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -640,28 +826,56 @@ export class MongoDBInstrumentation extends InstrumentationBase {
    * Ends a created span.
    * @param span The created span to end.
    * @param resultHandler A callback function.
+   * @param connectionId: The connection ID of the Command response.
    */
-  private _patchEnd(span: Span, resultHandler: Function): Function {
+  private _patchEnd(
+    span: Span | undefined,
+    resultHandler: Function,
+    connectionId?: number,
+    commandType?: string
+  ): Function {
     // mongodb is using "tick" when calling a callback, this way the context
     // in final callback (resultHandler) is lost
     const activeContext = context.active();
     const instrumentation = this;
     return function patchedEnd(this: {}, ...args: unknown[]) {
       const error = args[0];
-      if (error instanceof Error) {
-        span.setStatus({
-          code: SpanStatusCode.ERROR,
-          message: error.message,
-        });
-      } else {
-        const result = args[1] as CommandResult;
-        instrumentation._handleExecutionResult(span, result);
+      if (span) {
+        if (error instanceof Error) {
+          span?.setStatus({
+            code: SpanStatusCode.ERROR,
+            message: error.message,
+          });
+        } else {
+          const result = args[1] as CommandResult;
+          instrumentation._handleExecutionResult(span, result);
+        }
+        span.end();
       }
-      span.end();
 
       return context.with(activeContext, () => {
+        if (commandType === 'endSessions') {
+          instrumentation._connectionsUsage.add(-1, {
+            state: 'idle',
+            'pool.name': instrumentation._poolName,
+          });
+        }
         return resultHandler.apply(this, args);
       });
     };
   }
+  private setPoolName(options: any) {
```

Comment:
Comment to myself and others - I think until this pr: https://github.com/open-telemetry/opentelemetry-specification/pull/3050 is not approved, we can't approve the pool name issue. (This pr wants to define the pool.name to be the db.connection_string in case the DB doesn't supply pool name).